<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>947</width>
    <height>587</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="content">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>1</horstretch>
     <verstretch>1</verstretch>
    </sizepolicy>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QVBoxLayout" name="gameLayout"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>947</width>
     <height>20</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="tearOffEnabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Project"/>
    <addaction name="actionOpen_Project"/>
    <addaction name="separator"/>
    <addaction name="actionBuild"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuScene">
    <property name="title">
     <string>Scene</string>
    </property>
    <widget class="QMenu" name="menuCreate_Primative">
     <property name="title">
      <string>Create Primative</string>
     </property>
     <addaction name="actionPlane"/>
     <addaction name="actionCube"/>
     <addaction name="actionSphere"/>
    </widget>
    <addaction name="actionCreate_Empty_Entity"/>
    <addaction name="separator"/>
    <addaction name="menuCreate_Primative"/>
    <addaction name="actionCreate_Camera"/>
    <addaction name="actionCreate_Light"/>
   </widget>
   <widget class="QMenu" name="menuAssets">
    <property name="title">
     <string>Assets</string>
    </property>
    <widget class="QMenu" name="menuCreate">
     <property name="title">
      <string>Create</string>
     </property>
     <addaction name="actionFolder"/>
     <addaction name="separator"/>
     <addaction name="actionMaterial"/>
    </widget>
    <addaction name="menuCreate"/>
    <addaction name="separator"/>
    <addaction name="actionImport"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionGame"/>
    <addaction name="actionScene_Hierarchy"/>
    <addaction name="actionEntity_Details"/>
    <addaction name="actionAssets"/>
    <addaction name="actionConsole"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuScene"/>
   <addaction name="menuAssets"/>
   <addaction name="menuWindow"/>
  </widget>
  <widget class="CustomDockWidget" name="DockConsole">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="windowTitle">
    <string>Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_9">
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <widget class="QCheckBox" name="console_info_checkbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Info</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="console_warning_checkbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Warning</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="console_error_checkbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Error</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="CheckList" name="console_column_select">
         <property name="currentText">
          <string/>
         </property>
         <property name="sizeAdjustPolicy">
          <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
         </property>
         <property name="minimumContentsLength">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="CheckList" name="console_tag_select"/>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="console_search">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="maxLength">
          <number>1600</number>
         </property>
         <property name="placeholderText">
          <string>Search/Filter Logs</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="console_clear_console">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Clear Logs</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTableView" name="console_tableview">
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="CustomDockWidget" name="DockScene">
   <property name="windowTitle">
    <string>Scene</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_10">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLineEdit" name="scene_hierarchy_filter">
       <property name="placeholderText">
        <string>Search Entitys</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTreeView" name="scene_hierarchy_treeview">
       <property name="headerHidden">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="CustomDockWidget" name="DockEntityDetails">
   <property name="windowTitle">
    <string>Entity Details</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_11">
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QListView" name="listView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="CustomDockWidget" name="DockAssets">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="windowTitle">
    <string>Assets</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_13">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTreeView" name="assets_file_treeview">
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionNew_Project">
   <property name="text">
    <string>New Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen_Project">
   <property name="text">
    <string>Open Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionCreate_Empty_Entity">
   <property name="text">
    <string>Create Empty Entity</string>
   </property>
  </action>
  <action name="actionCreate_Camera">
   <property name="text">
    <string>Create Camera</string>
   </property>
  </action>
  <action name="actionCreate_Light">
   <property name="text">
    <string>Create Light</string>
   </property>
  </action>
  <action name="actionPlane">
   <property name="text">
    <string>Plane</string>
   </property>
  </action>
  <action name="actionCube">
   <property name="text">
    <string>Cube</string>
   </property>
  </action>
  <action name="actionSphere">
   <property name="text">
    <string>Sphere</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="text">
    <string>Import</string>
   </property>
  </action>
  <action name="actionFolder">
   <property name="text">
    <string>Folder</string>
   </property>
  </action>
  <action name="actionMaterial">
   <property name="text">
    <string>Material</string>
   </property>
  </action>
  <action name="actionGame">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Game</string>
   </property>
  </action>
  <action name="actionScene_Hierarchy">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scene Hierarchy</string>
   </property>
  </action>
  <action name="actionEntity_Details">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Entity Details</string>
   </property>
  </action>
  <action name="actionAssets">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Assets</string>
   </property>
  </action>
  <action name="actionConsole">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Console</string>
   </property>
  </action>
  <action name="actionBuild">
   <property name="text">
    <string>Build</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CustomDockWidget</class>
   <extends>QDockWidget</extends>
   <header location="global">customdockwidget.h</header>
   <container>1</container>
   <slots>
    <signal>dockClosed()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>CheckList</class>
   <extends>QComboBox</extends>
   <header location="global">checklist.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
